### DISCUSSION TITLE
Specifying keys with a specific key id (`kid`) value

### DISCUSSION THREAD

**Discussion by AlexGoris-KasparSolutions** [keycloak/keycloak#12131] [Q&A]
I've read in the [docs ](https://github.com/keycloak/keycloak-documentation/blob/main/server_admin/topics/realms/keys.adoc)how to add new keys, but as far as I can see, these get a key id which gets automatically defined by Keycloak upon creation.

Is it possible when adding new keys to assign them a custom key id value?

To be clear: I'm talking about the `kid` value which eventually is shown in the jwks json object served on the `jwks_uri` url.

Background: I'm trying to integrate with a 3rd party identity provider, through OpenID Connect. I need to use a signed jwt for authentication, but they require the use of a fixed key id value for their acceptance environment (in production, they'll collect the keys of my keycloak instance from the jwks_uri object).

**Comment by zloom** [comment]
I found a 2 ways:
1) Can be done in operator with realm import https://github.com/keycloak/keycloak/blob/main/operator/src/test/resources/example-realm.yaml#L1306
2) there is kc_component table in database it contains following JSON 
`{"entityVersion": 1, "fConfig": {"kid": ["432a3dd0-79b8-4806-ad2e-89b8df14841d"], .... "fProviderType": "org.keycloak.keys.KeyProvider"}`
You need to specify kid prop on this json add if it is missing.

**Reply by vishrantgupta** [reply]
This is not the best possible solution, @keycloak should look into this. @AlexGoris-KasparSolutions did you find any solution to this?

This topic is also related to this https://stackoverflow.com/questions/55752859/should-a-jwks-url-be-hosted-by-the-service-provider-or-the-openid-provider

**Reply by atrujillofalcon** [reply]
This is not implemented yet? Would very helpful

**Reply by zioerjens** [reply]
agreed, would be very helpful
